cppcache/include/geode/DataInput.hpp

```diff
@@ -578,6 +578,24 @@ class _GEODE_EXPORT DataInput {
     }
   }
 
+  template <typename T>
+  std::vector<T> readArray() {
+    int arrayLen = readArrayLen();
+    if (arrayLen < 0) {
+      return std::vector<T>(0);
+    }
+    std::vector<T> objArray(arrayLen);
```

Comment:
This constructor will allocate `arrayLen` elements with `T`'s default constructor.

Reply:
Allocate at the top and `reserve` later if `arrayLen > 0`.